src/components/Editor.js

```diff
@@ -382,6 +383,45 @@ const Editor = React.createClass({
     showMenu(e, items);
   },
 
+  traverseResultsPrev(e, ctx) {
+    e.stopPropagation();
+    e.preventDefault();
+    const { query } = this.state;
+    findPrev(ctx, query);
+  },
+
+  traverseResultsNext(e, ctx) {
+    e.stopPropagation();
+    e.preventDefault();
+    const { query } = this.state;
+    findNext(ctx, query);
+  },
+
+  onCursorActivity(ctx) {
+    const { searchOverlayMode } = this.state;
+    if (searchOverlayMode) {
+      if (!ctx.cm.somethingSelected()) {
+        removeOverlay(ctx);
+        this.setState({
+          searchOverlayMode: !searchOverlayMode,
+          query: ""
+        });
+      }
+    }
```

Comment:
couple things:

1. this seems like a mechanism for removing the search overlay mode. Is that correct?
2. could we also do a serch from here when the user lifts the mouse up
3. the IFs could be collapsed into one `if (searchOverlayMode && !ctx.cm.somethingSelected()) {}`
4. the searchOverlayMode would always be true, so in this case lets just do false... seems more honest